listener/avalon_listener/tcs_work_order_handler_sync.py

```diff
@@ -0,0 +1,248 @@
+# Copyright 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import json
+import logging
+from error_code.error_status import WorkOrderStatus
+from error_code.enclave_error import EnclaveError
+from avalon_sdk.direct.jrpc.jrpc_util import JsonRpcErrorCode
+from avalon_sdk.work_order.work_order_request_validator \
+    import WorkOrderRequestValidator
+from utility.hex_utils import is_valid_hex_str
+
+from jsonrpc.exceptions import JSONRPCDispatchException
+
+# ZeroMQ for sync workorder processing
+import zmq
+context = zmq.Context()
+# -------------------------------------------
+
+logger = logging.getLogger(__name__)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+
+class TCSWorkOrderHandler:
```

Comment:
Did you mean TCSWorkOrderHandlerSync ? Seems like there is lot of common code for TCSWorkOrderHandlerSync and TCSWorkOrderHandler. You could try to keep existing TCSWorkOrderHandler class in order to reuse the code and only selective change the functions related to synchronous work order handling like WorkOrderSubmit() based on config flag. Another option is you could make TCSWorkOrderHandler as a base class and derive TCSWorkOrderHandlerSync from TCSWorkOrderHandler.

Reply:
Okay.Updated in the latest commit. I have derived TCSWorkOrderHandlerSync from TCSWorkOrderHandler.